client/web/src/repo/blob/codemirror/token-selection/facets.ts

```diff
@@ -0,0 +1,30 @@
+import { Facet } from '@codemirror/state'
+import { Remote } from 'comlink'
+import * as H from 'history'
+
+import { FlatExtensionHostAPI } from '@sourcegraph/shared/src/api/contract'
+
+import { BlobInfo } from '../../Blob'
+
+export const textDocumentImplemenationSupport = Facet.define<boolean, boolean>({
+    combine: props => props[0],
+})
+export const uriFacet = Facet.define<string, string>({
+    combine: props => props[0],
+})
+export const blobInfoFacet = Facet.define<BlobInfo, BlobInfo>({
+    combine: props => props[0],
+})
+export const codeintelFacet = Facet.define<Remote<FlatExtensionHostAPI>, Remote<FlatExtensionHostAPI>>({
```

Comment:
```suggestion
export const codeIntelFacet = Facet.define<Remote<FlatExtensionHostAPI>, Remote<FlatExtensionHostAPI>>({
```
nit: camelCasing the variable name for consistency with the rest of the codebase

Reply:
Done.